server/src/main/java/org/apache/druid/server/audit/AuditManagerConfig.java

```diff
@@ -19,11 +19,20 @@
 
 package org.apache.druid.server.audit;
 
-import com.google.inject.Provider;
-import org.apache.druid.audit.AuditManager;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
 
-public interface AuditManagerProvider extends Provider<AuditManager>
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = SQLAuditManagerConfig.class)
```

Comment:
default should be no-op?

Reply:
If we default to no-op, some existing functionality would misbehave:
- change history dialogs in compaction config, load rules, etc